src/File.hs

```diff
@@ -29,6 +28,19 @@ data ExFile = ExFile {
 
 dotfileRegEx = "^\\/?(?:\\w+\\/)*(\\.\\w+)" :: String
 
+{-|
+    @brief Given a filepath and a recursive flag, dispatches
+    the correct file finder.
+    @throws IOException if file/dir does not exist
+    @param filepath String
+    @param recursive Bool
+    @returns list of files and their contents
+-}
+dispatchFileExtraction :: FilePath -> Bool -> IO [ExFile]
```

Comment:
Huh, I guess everyone loves dispatchers.

Reply:
I guess because conditionals in haskell are gross

Reply:
Yea it just seems cleaner this way